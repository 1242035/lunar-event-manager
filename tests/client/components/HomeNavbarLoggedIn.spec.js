import React from 'react';
import { shallow } from 'enzyme';
import ConnectedHomeNavbarLoggedIn from '../../../client/src/components/HomeNavbarLoggedIn';
import mockStore from '../mocks/storeMock';
import { mockAxios } from '../mocks/axiosMock';

describe('HomeNavbarLoggedIn', () => {
  let wrapper;
  let store;
  mockAxios();

  beforeAll(() => {
    store = mockStore();
    localStorage.setItem('user.token', 'token');
    localStorage.setItem('user.admin', 'true');
  });

  afterAll(() => {
    localStorage.setItem('user.token', 'undefined');
    localStorage.setItem('user.admin', 'undefined');
  });

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<ConnectedHomeNavbarLoggedIn store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render connected component', () => {
    wrapper = shallow(<ConnectedHomeNavbarLoggedIn store={store} />);
    expect(wrapper.length).toEqual(1);
  });
});
