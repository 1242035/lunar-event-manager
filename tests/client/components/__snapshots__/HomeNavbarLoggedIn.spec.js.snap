// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeNavbarLoggedIn Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeNavbarLoggedIn
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Link
          className="io-brand"
          href={true}
          replace={false}
          to="/"
        >
          <div
            className="io-text"
          >
            EventsManagerIO
          </div>
          <img
            alt=""
            className="io-img"
            src="test-file-stub"
          />
        </Link>,
        <div
          className="io-middle io-start"
        >
          <a
            href="/#footer"
          >
            CONTACT
          </a>
          <a
            href="/#about"
          >
            ABOUT
          </a>
        </div>,
        <div
          className="io-end"
        >
          <Link
            className="io-text"
            href={true}
            id="goto-discover"
            replace={false}
            to="/discover"
          >
            DISCOVER
          </Link>
          <Link
            className="io-text"
            href={true}
            replace={false}
            to="/profile"
          >
            PROFILE
          </Link>
        </div>,
      ],
      "className": "io-navbar io-fixed-top",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="io-text"
            >
              EventsManagerIO
            </div>,
            <img
              alt=""
              className="io-img"
              src="test-file-stub"
            />,
          ],
          "className": "io-brand",
          "href": true,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EventsManagerIO",
              "className": "io-text",
            },
            "ref": null,
            "rendered": "EventsManagerIO",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "io-img",
              "src": "test-file-stub",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="/#footer"
            >
              CONTACT
            </a>,
            <a
              href="/#about"
            >
              ABOUT
            </a>,
          ],
          "className": "io-middle io-start",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CONTACT",
              "href": "/#footer",
            },
            "ref": null,
            "rendered": "CONTACT",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ABOUT",
              "href": "/#about",
            },
            "ref": null,
            "rendered": "ABOUT",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="io-text"
              href={true}
              id="goto-discover"
              replace={false}
              to="/discover"
            >
              DISCOVER
            </Link>,
            <Link
              className="io-text"
              href={true}
              replace={false}
              to="/profile"
            >
              PROFILE
            </Link>,
          ],
          "className": "io-end",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "DISCOVER",
              "className": "io-text",
              "href": true,
              "id": "goto-discover",
              "replace": false,
              "to": "/discover",
            },
            "ref": null,
            "rendered": "DISCOVER",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "PROFILE",
              "className": "io-text",
              "href": true,
              "replace": false,
              "to": "/profile",
            },
            "ref": null,
            "rendered": "PROFILE",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Link
            className="io-brand"
            href={true}
            replace={false}
            to="/"
          >
            <div
              className="io-text"
            >
              EventsManagerIO
            </div>
            <img
              alt=""
              className="io-img"
              src="test-file-stub"
            />
          </Link>,
          <div
            className="io-middle io-start"
          >
            <a
              href="/#footer"
            >
              CONTACT
            </a>
            <a
              href="/#about"
            >
              ABOUT
            </a>
          </div>,
          <div
            className="io-end"
          >
            <Link
              className="io-text"
              href={true}
              id="goto-discover"
              replace={false}
              to="/discover"
            >
              DISCOVER
            </Link>
            <Link
              className="io-text"
              href={true}
              replace={false}
              to="/profile"
            >
              PROFILE
            </Link>
          </div>,
        ],
        "className": "io-navbar io-fixed-top",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="io-text"
              >
                EventsManagerIO
              </div>,
              <img
                alt=""
                className="io-img"
                src="test-file-stub"
              />,
            ],
            "className": "io-brand",
            "href": true,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "EventsManagerIO",
                "className": "io-text",
              },
              "ref": null,
              "rendered": "EventsManagerIO",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "io-img",
                "src": "test-file-stub",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="/#footer"
              >
                CONTACT
              </a>,
              <a
                href="/#about"
              >
                ABOUT
              </a>,
            ],
            "className": "io-middle io-start",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CONTACT",
                "href": "/#footer",
              },
              "ref": null,
              "rendered": "CONTACT",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ABOUT",
                "href": "/#about",
              },
              "ref": null,
              "rendered": "ABOUT",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="io-text"
                href={true}
                id="goto-discover"
                replace={false}
                to="/discover"
              >
                DISCOVER
              </Link>,
              <Link
                className="io-text"
                href={true}
                replace={false}
                to="/profile"
              >
                PROFILE
              </Link>,
            ],
            "className": "io-end",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "DISCOVER",
                "className": "io-text",
                "href": true,
                "id": "goto-discover",
                "replace": false,
                "to": "/discover",
              },
              "ref": null,
              "rendered": "DISCOVER",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "PROFILE",
                "className": "io-text",
                "href": true,
                "replace": false,
                "to": "/profile",
              },
              "ref": null,
              "rendered": "PROFILE",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
